File: src/api/endpoints/staff/StaffEndpoint.ts
import { Logger } from "ts-log";

import { HeaderProvider } from "../../../Func";
import ApiResponse from "../../ApiResponse";
import ApiEndpoint from "../ApiEndpoint";
import TokenManager from "../auth/TokenManager";
import DefaultApiConfig from "../DefaultApiConfig";
import MedalInformation from "./response/MedalInformation";

/**
 * Endpoints for interacting with staff.
 * */
export default class StaffEndpoint extends ApiEndpoint {
  constructor(config: DefaultApiConfig, tokenManager: TokenManager, log?: Logger, headerProvider?: HeaderProvider) {
    super(config, tokenManager, log, headerProvider);
  }

  protected override endpoint(): string {
    return "staff";
  }

  /**
   * Gets detailed information about medals.
   * */
  public async medalsInformation(): Promise<ApiResponse<MedalInformation[]>> {
    return await this.getRequest<MedalInformation[]>("/meta/medals");
  }
}

File: src-dotnet/api/endpoints/staff/StaffEndpoint.cs
namespace MedrunnerApiClient.Api.Endpoints.Staff
{
    /// <summary>
    /// Endpoints for interacting with staff.
    /// </summary>
    public class StaffEndpoint : ApiEndpoint
    {
        public StaffEndpoint(ApiConfig config, TokenManager tokenManager, ILogger logger)
            : base(config, tokenManager, logger)
        {
        }

        protected override string Endpoint => "staff";

        /// <summary>
        /// Gets detailed information about medals.
        /// </summary>
        public async Task<ApiResponse<List<Api.Endpoints.Staff.Response.MedalInformation>>> MedalsInformationAsync()
        {
            return await GetRequestAsync<List<Api.Endpoints.Staff.Response.MedalInformation>>("/meta/medals");
        }
    }
}
